ring/spread_minimizing_token_generator.go

```diff
@@ -331,21 +310,18 @@ func (t *SpreadMinimizingTokenGenerator) generateTokensByInstanceID() map[int]To
 		heap.Push(&instanceQueue, newRingInstanceOwnershipInfo(i, currInstanceOwnership))
 	}
 
-	return tokensByInstanceID
+	return tokensByInstanceID, nil
 }
 
 func (t *SpreadMinimizingTokenGenerator) CanJoin(instances map[string]InstanceDesc) error {
 	if !t.canJoinEnabled {
 		return nil
 	}
 
-	prevInstance, err := previousInstance(t.instance)
-	if err != nil {
-		if errors.Is(err, errorNoPreviousInstance) {
```

Comment:
Could you please check whether we still need `errorNoPreviousInstance`? I think that at this point it should be removed.

Reply:
This PR removed `errorNoPreviousInstance` variable.